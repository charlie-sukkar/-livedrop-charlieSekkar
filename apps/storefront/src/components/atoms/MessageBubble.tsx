import React from 'react';

interface MessageBubbleProps {
  message: string;
  isUser?: boolean;
  timestamp?: string;
  citation?: string;
}

export const MessageBubble: React.FC<MessageBubbleProps> = ({ 
  message, 
  isUser = false, 
  timestamp,
  citation 
}) => {
  return (
    <div className={`flex ${isUser ? 'justify-end' : 'justify-start'} mb-4`}>
      <div 
        className={`
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          ${isUser 
            ? 'bg-blue-600 text-white rounded-br-none' 
            : 'bg-gray-200 text-gray-900 rounded-bl-none'
          }
        `}
        role="text"
        aria-label={`${isUser ? 'You' : 'Support'} said: ${message}`}
      >
        <p className="text-sm whitespace-pre-wrap">{message}</p>
        
        {(timestamp || citation) && (
          <div className={`flex justify-between items-center mt-1 text-xs ${
            isUser ? 'text-blue-100' : 'text-gray-500'
          }`}>
            {timestamp && <span>{timestamp}</span>}
            {citation && (
              <span className="ml-2 bg-black bg-opacity-20 px-1 rounded">
                {citation}
              </span>
            )}
          </div>
        )}
      </div>
    </div>
  );
};