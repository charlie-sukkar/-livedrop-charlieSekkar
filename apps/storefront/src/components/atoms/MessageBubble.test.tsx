import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { MessageBubble } from './MessageBubble';

describe('MessageBubble Component', () => {
  const testMessage = 'Hello, this is a test message';

  it('renders message content', () => {
    render(<MessageBubble message={testMessage} />);
    expect(screen.getByText(testMessage)).toBeInTheDocument();
  });

  it('user message has blue background', () => {
    render(<MessageBubble message={testMessage} isUser={true} />);
    const bubble = screen.getByRole('text');
    expect(bubble).toHaveClass('bg-blue-600');
  });

  it('support message has gray background', () => {
    render(<MessageBubble message={testMessage} isUser={false} />);
    const bubble = screen.getByRole('text');
    expect(bubble).toHaveClass('bg-gray-200');
  });

  it('shows timestamp when provided', () => {
    const timestamp = '2:30 PM';
    render(<MessageBubble message={testMessage} timestamp={timestamp} />);
    expect(screen.getByText(timestamp)).toBeInTheDocument();
  });

  it('shows citation when provided', () => {
    const citation = 'Documentation';
    render(<MessageBubble message={testMessage} citation={citation} />);
    expect(screen.getByText(citation)).toBeInTheDocument();
  });

  it('has accessible label', () => {
    render(<MessageBubble message={testMessage} isUser={true} />);
    const bubble = screen.getByRole('text');
    expect(bubble).toHaveAttribute('aria-label', `You said: ${testMessage}`);
  });

  it('message text wraps properly', () => {
    render(<MessageBubble message={testMessage} />);
    const messageElement = screen.getByText(testMessage);
    expect(messageElement).toHaveClass('whitespace-pre-wrap');
  });
});