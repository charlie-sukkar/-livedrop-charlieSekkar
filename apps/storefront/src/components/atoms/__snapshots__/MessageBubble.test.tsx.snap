// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessageBubble > matches snapshot for minimal message 1`] = `
<div
  class="flex justify-start mb-4"
>
  <div
    aria-label="Support said: OK"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-gray-200 text-gray-900 rounded-bl-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      OK
    </p>
  </div>
</div>
`;

exports[`MessageBubble > matches snapshot for support message with all props 1`] = `
<div
  class="flex justify-start mb-4"
>
  <div
    aria-label="Support said: Support response"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-gray-200 text-gray-900 rounded-bl-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      Support response
    </p>
    <div
      class="flex justify-between items-center mt-1 text-xs text-gray-500"
    >
      <span>
        2:31 PM
      </span>
      <span
        class="ml-2 bg-black bg-opacity-20 px-1 rounded"
      >
        Knowledge Base
      </span>
    </div>
  </div>
</div>
`;

exports[`MessageBubble > matches snapshot for user message with all props 1`] = `
<div
  class="flex justify-end mb-4"
>
  <div
    aria-label="You said: User question"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-blue-600 text-white rounded-br-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      User question
    </p>
    <div
      class="flex justify-between items-center mt-1 text-xs text-blue-100"
    >
      <span>
        2:30 PM
      </span>
      <span
        class="ml-2 bg-black bg-opacity-20 px-1 rounded"
      >
        User Guide
      </span>
    </div>
  </div>
</div>
`;

exports[`MessageBubble Component > matches snapshot 1`] = `
<div>
  <div
    class="flex justify-end mb-4"
  >
    <div
      aria-label="You said: Hello, this is a test message"
      class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-blue-600 text-white rounded-br-none
        "
      role="text"
    >
      <p
        class="text-sm whitespace-pre-wrap"
      >
        Hello, this is a test message
      </p>
    </div>
  </div>
</div>
`;

exports[`MessageBubble Component > matches snapshot for minimal message 1`] = `
<div
  class="flex justify-start mb-4"
>
  <div
    aria-label="Support said: OK"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-gray-200 text-gray-900 rounded-bl-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      OK
    </p>
  </div>
</div>
`;

exports[`MessageBubble Component > matches snapshot for support message 1`] = `
<div
  class="flex justify-start mb-4"
>
  <div
    aria-label="Support said: How can I help you today?"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-gray-200 text-gray-900 rounded-bl-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      How can I help you today?
    </p>
    <div
      class="flex justify-between items-center mt-1 text-xs text-gray-500"
    >
      <span>
        2:31 PM
      </span>
      <span
        class="ml-2 bg-black bg-opacity-20 px-1 rounded"
      >
        Support KB
      </span>
    </div>
  </div>
</div>
`;

exports[`MessageBubble Component > matches snapshot for user message 1`] = `
<div
  class="flex justify-end mb-4"
>
  <div
    aria-label="You said: Hello there"
    class="
          max-w-xs lg:max-w-md px-4 py-2 rounded-lg
          bg-blue-600 text-white rounded-br-none
        "
    role="text"
  >
    <p
      class="text-sm whitespace-pre-wrap"
    >
      Hello there
    </p>
    <div
      class="flex justify-between items-center mt-1 text-xs text-blue-100"
    >
      <span>
        2:30 PM
      </span>
      <span
        class="ml-2 bg-black bg-opacity-20 px-1 rounded"
      >
        User Guide
      </span>
    </div>
  </div>
</div>
`;
